src/utils/text.cc

```diff
@@ -171,14 +170,7 @@ void mbs_write_wrapped(ostream & out, const string & text,
     bytes_read = mbrtowc (&wc, s, s_bytes, &shift_state);
     if (bytes_read > 0)
```

Comment:
Looking at the history, I'd say the author simply forgot that ```bytes_read``` is unsigned and wanted to catch the case in the trailing else. I'd simply change to:
```
if ( bytes_read > 0 && bytes_read < (size_t) -2 )

```